kdl_parser/include/kdl_parser/kdl_parser.hpp

```diff
@@ -77,6 +77,7 @@ bool treeFromString(const std::string & xml, KDL::Tree & tree);
  * \param tree The resulting KDL Tree
  * returns true on success, false on failure
  */
+[[deprecated("Use treeFromString instead")]]
```

Comment:
So, it turns out that we are using treeFromXml internally in https://github.com/ros2/kdl_parser/blob/ce5323ce412a679b37623ae2bcc08a074cf8410e/kdl_parser/src/kdl_parser.cpp#L161 and https://github.com/ros2/kdl_parser/blob/ce5323ce412a679b37623ae2bcc08a074cf8410e/kdl_parser/src/kdl_parser.cpp#L182 , which is triggering the deprecation warning.  For this to go in, we'll need to fix those as well to use something else.

Reply:
Addressed in #7 where the deprecation is fully resolved. I'll move those changes to this PR.